sqlmesh/dbt/basemodel.py

```diff
@@ -243,7 +243,6 @@ def render_config(self: BMC, context: DbtContext) -> BMC:
                 rendered._dependencies = rendered._dependencies.union(
                     model.render_config(context)._dependencies
                 )
-                rendered._dependencies.refs.discard(dependency)
```

Comment:
This should resolve https://github.com/TobikoData/sqlmesh/issues/668, but might need to talk it through some more, cc: @crericha.